src/page-modules/contact/lines/use-lines.ts

```diff
@@ -10,9 +10,36 @@ export const useLines = () => {
     isLoading,
   } = useSWR<Line[]>('/api/contact/lines', swrFetcher);
 
-  const getLinesByMode = (mode: TransportModeType): Line[] => {
+  const getLinesByMode = (transportMode: TransportModeType): Line[] => {
     if (!lines) return [];
-    return lines.filter((line) => line.transportMode === mode);
+
+    switch (transportMode) {
+      case 'bus':
+        return lines.filter((line) => line.transportMode === transportMode);
+
+      case 'expressboat':
+        return lines.filter(
+          (line) =>
+            line.transportMode === 'water' &&
+            (line.transportSubmode === 'highSpeedPassengerService' ||
```

Comment:
This could probably be made cleaner by using e.g. `Array.includes`, but not that big of a deal for me.